tests/foreman/api/test_product.py

```diff
@@ -399,3 +403,46 @@ def test_positive_filter_product_list(self):
         self.assertGreater(len(rh_products), 1)
         self.assertNotIn(product.name, [prod.name for prod in rh_products])
         self.assertIn('Red Hat Beta', [prod.name for prod in rh_products])
+
+    @tier2
+    def test_positive_assign_http_proxy_to_products(self):
+        """Assign http_proxy to Products.
+
+        :id: c9d23aa1-3325-4abd-a1a6-d5e75c12b08a
+
+        :expectedresults: HTTP Proxy is assigned to all repos present
+            in Products.
+
+        :CaseImportance: Critical
+        """
+
+        http_proxy_url = '{}:{}'.format(
+            gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999))
+        http_proxy = entities.HTTPProxy(
+            name=gen_string('alpha', 15),
+            url=http_proxy_url,
+            organization=[self.org.id]
+        ).create()
+        product_a = entities.Product(organization=self.org).create()
+        product_b = entities.Product(organization=self.org).create()
+        repo_a = entities.Repository(
+            product=product_a,
+            http_proxy_policy='none',
+        ).create()
+        repo_b = entities.Repository(
+            product=product_b,
+            http_proxy_policy='global_default_http_proxy',
+        ).create()
+        # Add http_proxy to products
+        entities.ProductBulkAction().http_proxy(
+            data={
+                "ids": [product_a.id, product_b.id],
+                "http_proxy_policy": "use_selected_http_proxy",
+                "http_proxy_id": http_proxy.id
+            })
+        updated_repo_a = entities.Repository(id=repo_a.id).read()
```

Comment:
```suggestion
        repo_a = entities.Repository(id=repo_a.id).read()
```
Updated repo is slightly has different mean , please use as above to avoid naming confusion.

Reply:
Updated PR.